packages/mobile/src/exchange/CeloGoldHistoryChart.tsx

```diff
@@ -0,0 +1,295 @@
+  const dollarsToLocal = useCallback(
+    (amount) => convertDollarsToLocalAmount(amount, localExchangeRate),
+    [localExchangeRate]
+  )
+  const [range] = useState(30 * 24 * 60 * 60 * 1000) // 30 days
+  const exchangeHistory = useSelector(exchangeHistorySelector)
+
+  if (!exchangeHistory.celoGoldExchangeRates.length) {
+    return <Loader />
+  }
+
+  const groupedExchangeHistory = _.groupBy(exchangeHistory.celoGoldExchangeRates, calculateGroup)
+  const latestGroup = calculateGroup(
+    exchangeHistory.celoGoldExchangeRates[exchangeHistory.celoGoldExchangeRates.length - 1]
+  )
+  const chartData = _.range(
+    Math.min(CHART_POINTS_NUMBER - 1, Object.keys(groupedExchangeHistory).length),
+    0,
+    -1
+  ).map((i) => {
+    const group = groupedExchangeHistory[latestGroup - i + 1]
+    const localAmount = dollarsToLocal(
+      group ? _.meanBy(group, (er) => parseFloat(er.exchangeRate)) : 0
+    )
+    return {
+      amount: localAmount ? localAmount.toNumber() : 0,
+      displayValue: localAmount ? displayLocalCurrency(localAmount) : '',
+    }
+  })
+
+  const currentGoldRateInLocalCurrency = dollarsToLocal(goldToDollars(1))
+  const oldestGoldRateInLocalCurrency = chartData[0].amount
+  if (oldestGoldRateInLocalCurrency == null || currentGoldRateInLocalCurrency == null) {
+    return <Loader />
+  }
+  // We need displayValue to show min/max on the chart. In case the
+  // current value is min/max we do not need to show it once again,
+  // therefor displayValue = ''
+  chartData.push({ amount: currentGoldRateInLocalCurrency.toNumber(), displayValue: '' })
+  const rateChange = currentGoldRateInLocalCurrency.minus(oldestGoldRateInLocalCurrency)
+  const rateChangeInPercentage = currentGoldRateInLocalCurrency
+    .div(oldestGoldRateInLocalCurrency)
+    .minus(1)
+    .multipliedBy(100)
+  const RenderPoint = renderPointOnChart(chartData)
+
+  const values = chartData.map((el) => el.amount)
+  const min = Math.min(...values)
+  const max = Math.max(...values)
+  let domain = null
+  // ensure that vertical chart range is at least CHART_MIN_VERTICAL_RANGE
+  if (max - min < CHART_MIN_VERTICAL_RANGE) {
+    const offset = CHART_MIN_VERTICAL_RANGE - (max - min) / 2
+    domain = { y: [min - offset, max + offset] }
+  }
+
+  return (
+    <View style={styles.container} testID={testID}>
+      <View style={styles.goldPrice}>
+        <View>
+          <Text style={styles.goldPriceTitle}>Gold Price</Text>
+        </View>
+        <View style={styles.goldPriceValues}>
+          <Text style={styles.goldPriceCurrentValue}>
+            {displayLocalCurrency(currentGoldRateInLocalCurrency)}
+          </Text>
+          <Text style={styles.goldPriceChange}>
+            {rateChange.gt(0) ? '▴' : '▾'} {rateChange.toFixed(2)} (
+            {rateChangeInPercentage.toFixed(2)}%)
+          </Text>
+        </View>
+      </View>
+      <VictoryGroup
+        domainPadding={{ y: [30, 30], x: [5, 5] }}
```

Comment:
Nit: if this is constant maybe define it outside the render function for efficiency?